### PR TITLE
feat!: Update dependencies and encapsulate client construction.

### PR DESCRIPTION
**Pull request by kinyoklion**
Updates to latest OpenFeature SDK.
Updates to latest LD SDK. Addresses #34 

Adds support for OpenFeature events.
Adds support for provider shutdown. Addresses #15.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
       matrix:
         variations: [
           {os: ubuntu-latest, node: latest},
-          {os: ubuntu-latest, node: 14},
+          {os: ubuntu-latest, node: 16},
```
Comment (kinyoklion) [inline .github/workflows/ci.yml:19]: OpenFeature SDK requires node 16.

File: package.json [hunk 2]
```diff
@@ -20,11 +20,11 @@
   ],
   "license": "Apache-2.0",
   "peerDependencies": {
-    "@openfeature/js-sdk": "^1.0.0",
```
Comment (kinyoklion) [inline package.json:23]: The OpenFeature SDK was renamed to make it clear that their is a client and server version.

File: src/LaunchDarklyProvider.ts [hunk 3]
```diff
@@ -31,20 +37,67 @@ function wrongTypeResult<T>(value: T): ResolutionDetails<T> {
...
+
   /**
    * Construct a {@link LaunchDarklyProvider}.
    * @param client The LaunchDarkly client instance to use.
    */
-  constructor(private readonly client: LDClient, options: LaunchDarklyProviderOptions = {}) {
+  constructor(sdkKey: string, options: LDOptions = {}) {
     if (options.logger) {
-      this.logger = options.logger;
+      this.logger = new SafeLogger(options.logger, basicLogger({ level: 'info' }));
```
Comment (kinyoklion) [inline src/LaunchDarklyProvider.ts:65]: This is just a fix, it should have had this already.

File: src/LaunchDarklyProvider.ts [hunk 4]
```diff
@@ -31,20 +37,67 @@ function wrongTypeResult<T>(value: T): ResolutionDetails<T> {
...
     } else {
       this.logger = basicLogger({ level: 'info' });
     }
+    try {
+      this.client = init(sdkKey, {
```
Comment (kinyoklion) [inline src/LaunchDarklyProvider.ts:70]: There are a few cases where the constructor fails. I am capturing it and letting the OF SDK handle it. That seems most in the spirit of the functionality. Maybe a little less convenient though.

**Review by keelerm84 — APPROVED** [review]

**Pull request merged**